scripts/Muon/GUI/Common/muon_pair.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import (absolute_import, division, print_function)
+
+from Muon.GUI.Common.muon_workspace import MuonWorkspace
+
+
+class MuonPair(object):
+    """Simple struct to store information on a detector group pair.
+
+    The name is set at initialization and after that cannot be changed.
```

Comment:
Yes again the answer is consistent with the `MuonGroup`; the name of the pair is used as a dictionary key; the value of which in the `MuonPair` object. To have the dictionary key and the `pair.name` get out of sync by allowing the name change would just remove one aspect of the "user contract" I have tried to establish for my core code.

More specifically; there could be bugs introduced in future coder whose authors would otherwise not be made aware that they should not change the name. 

This behavior is also unit tested so it is something we can rely on to be maintained.